flink/v1.18/flink/src/test/java/org/apache/iceberg/flink/TestHelpers.java

```diff
@@ -416,7 +416,7 @@ private static void assertAvroEquals(
         Collection<?> expectedArrayData = (Collection<?>) expected;
         ArrayData actualArrayData;
         try {
-          actualArrayData = (ArrayData) actual;
+          actualArrayData = convertToArray(actual);
```

Comment:
is this a behavior change from 1.18? it seems like a breaking change to me.

Reply:
only on 1.18 and it only affects the unit tests

Reply:
I know you only updated the unit test. but it suggests that Flink `RowData` field has changed from `ArrayData` to the raw type array (like `int[]`). I am trying to understand if it has any impact on compatibility.

Reply:
> I know you only updated the unit test. but it suggests that Flink `RowData` field has changed from `ArrayData` to the raw type array (like `int[]`). I am trying to understand if it has any impact on compatibility.

Thanks! cc: @pvary

Reply:
Based on @PrabhuJoseph's [comment](https://github.com/apache/iceberg/issues/8930#issuecomment-1806668299) the changes are caused by https://issues.apache.org/jira/browse/FLINK-33523.
Based on the jira, this is an expected general fix in Flink 1.18.

Reply:
With the clarify from the discussion thread that @pvary started, I am good with following up on this one separately when Flink 1.18.1 is released